processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -352,9 +352,15 @@ private void initPredicate()
           if (hasLowerBound()) {
             final Long lowerLong = GuavaUtils.tryParseLong(lower);
             if (lowerLong == null) {
-              // Unparseable values fall before all actual numbers, so all numbers will match the lower bound.
-              hasLowerLongBound = false;
-              lowerLongBound = 0L;
+              final Float lowerFloat = Floats.tryParse(lower);
```

Comment:
Float's precision is too bad, there is no point to use it if we are going to convert to long anyway. E. g. currently `100000001.0f` is converted to `100000000L` (please add such test, btw). Suggested to use at least `double`, or better `BigDecimal`.